homeassistant/components/sonarr/sensor.py

```diff
@@ -86,164 +97,71 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Sonarr sensors based on a config entry."""
-    sonarr: SonarrClient = hass.data[DOMAIN][entry.entry_id][DATA_SONARR]
-    host_config: PyArrHostConfiguration = hass.data[DOMAIN][entry.entry_id][
-        DATA_HOST_CONFIG
+    coordinators: dict[str, SonarrDataUpdateCoordinator] = hass.data[DOMAIN][
+        entry.entry_id
     ]
-    system_status: SystemStatus = hass.data[DOMAIN][entry.entry_id][DATA_SYSTEM_STATUS]
-    options: dict[str, Any] = dict(entry.options)
-
-    entities = [
+    async_add_entities(
         SonarrSensor(
-            sonarr,
-            host_config,
-            system_status,
-            entry.entry_id,
+            coordinators[coordinator_type],
             description,
```

Comment:
```suggestion
            description
```
Small formatting suggestion. If you remove the trailing comma, `black` might put all arguments on the same line.